packages/generator/src/parser/edmx-parser-v4.ts

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import { forceArray } from '../generator-utils';
+import {
+  EdmxEntitySet,
+  EdmxMetadata,
+  EdmxEntityType,
+  EdmxEnumType,
+  EdmxDerivedType,
+  EdmxComplexType
+} from './parser-types-v4';
+import {
+  parseEntityTypes,
+  parseFunctionImports,
+  parseComplexTypes
+} from './edmx-parser-common';
+import { stripNamespace, parseTypeName } from './parser-util';
+import { EdmxMetadataBase } from './parser-types-common';
+
+function joinTypesWithBaseTypes<T extends EdmxDerivedType>(
+  types: T[],
+  joinTypes: (type: T, baseType: T) => T
```

Comment:
I know this is all private., but could you create a type and then mark the two implementations `joinComplexTypes` and `joinEntityTypes` to have this type.

Reply:
I would prefer to keep it simple here. The function clearly states that we need something that takes two things of the same type and returns something of the same type. If I add an additional Type, this will be obfuscated, because I would have to check the type definition to know this.
I agree, that it would be nicer to somehow know that the existing joinXXX methods are the ones used for that, but I don't think it is worth changing that.

Reply:
Ok not critical and only private.